src/Sentry.AspNetCore.Blazor.WebAssembly/WebAssemblyHostBuilderExtensions.cs

```diff
@@ -28,10 +30,27 @@ public static WebAssemblyHostBuilder UseSentry(this WebAssemblyHostBuilder build
             blazorOptions.RequestBodyCompressionLevel = CompressionLevel.NoCompression;
             // Since the WebAssemblyHost is a client-side application
             blazorOptions.IsGlobalModeEnabled = true;
+            // If profiling enabled, disable it.
+            RemoveBlazorProfilingIntegration(blazorOptions);
         });
 
         return builder;
     }
+
+    private static void RemoveBlazorProfilingIntegration(SentryBlazorOptions options)
+    {
+        if (!options.IsProfilingEnabled)
+        {
+            return;
+        }
+
+        options.SetupLogging();
+        options.LogDebug("Detected Sentry profiling initialization in Blazor WebAssembly. " +
+                         "Sentry does not support Blazor WebAssembly profiling. Removing profiling integration. " +
+                         "Check https://github.com/getsentry/sentry-dotnet/issues/4506 for more information.");
+        // Ensure project doesn't have Profiling Integration
+        options.RemoveIntegration<ProfilingIntegration>();
```

Comment:
Nice - thanks @alexsohn1126 !

We could also add a unit test to `SentryWebHostBuilderExtensionsTests` to make sure that the integration gets removed.